### DISCUSSION TITLE
1st October Release : 2.78

### DISCUSSION THREAD

**Discussion by Aranud** [etlegacy/etlegacy#1667] [General]
@rmarquis @isRyven 3 months have passed since the 2.77 realease. To ensure a regular deployment (quarterly) it is therefore time to take care of making a new version! Do you have any changes or remarks to make to start this new version? Or do you think we should wait before we deploy anything?

**Comment by isRyven** [comment]
Sounds good. 

However, there seems to be some critical bug related to environment awareness feature, it seems sometimes it can lead to a infinite loop upon tracing something which is presumably called by https://github.com/etlegacy/etlegacy/blob/master/src/cgame/cg_draw.c#L3850-L3857, i managed to attach debugger to it during one of this cases where game became unresponsive, and i only briefly checked the stack trace, as i was not in mood to debug it, as i just wanted to clear out what exactly caused my game to stuck. Dunno if someone mentioned this already, dunno if this is due to 64bitness or anything else, but maybe you have some ideas? I could potentially try to debug it next time, but i am quite rusty. I could create ticket for this, but i am not sure if this is just my debug build, or some logical error in the code.

Sometimes you can't build the constructible if you are in prone and near the construction boxes (i experienced this during playing the braundorf, trying to fix the side gates), sometimes construction sound is not played.

But other than that, seems okay (not worse than previous release).

**Reply by Aranud** [reply]
I had corrected infinite loops previously which caused crashes on the TM server. They haven't had a problem since. I'm a taker if you still manage to reproduce the problem. I don't really like using a loop with the traces either ... maybe I'll add a number of iterations before exiting to make sure there is no problem like with the action activate.

I will look at the construction box on braundorf, I am sure @Aciz will be happy too if I correct the problem.

Thanks for your return !

**Reply by Aranud** [reply]
Alright, I did the required change for HUD icons.

About the construction box on Braundorf, I can't reproduce the issue 
![image](https://user-images.githubusercontent.com/5512525/121508375-d0a37880-c9e5-11eb-8fe5-22fe566f6a14.png)

Any tips ?

**Reply by isRyven** [reply]
Might be just a map issue, but if you aim far enough from the center to the edges no action would apply, while cursorhint with progress bar would still appear.

https://streamable.com/yhqvwi
(aiming and the far left of the boxes or top center, so basically on the edges, would cause the "bug", or rather just visual disparity)

Apparently, however, this case happened multiple times in battle, usually you try to hide yourself as much as possible sticking to the wall, and sometimes you are being pushed back on hit, so everything adds up: the distance and the angle you are aiming to build thing to hide from the shooting.

**Reply by isRyven** [reply]
Regarding the icons, it seems i had a bit old build, which was prior to your initial infinite loop fix, but it seems the initial fix also works, but i haven't played long enough yet, but i guess additional precautions won't be extra!

**Reply by rmarquis** [reply]
I've not had a close look at development for some time, so I've no real input to share here. But a new version sounds good, as to keep things rolling.

I've enabled the hook so Github discussions should show up in discord too.

**Reply by Aranud** [reply]
@isRyven it is a regression with changes done in Weapon_Engineer. The 1st trace hit the IOD trigger, but ignore it and 2nd hit an ent mover script .... and ignore it. The player in in the TOI, but we don't execute this part of code anymore (due to priority plier hit changes). It work well on vanilla. I'm trying to find the best solution to allow triggering the IOD ent.

**Reply by Aranud** [reply]
Could you have a try ? To ensure I didn't mess something ;)

**Reply by Aranud** [reply]
@rmarquis thank you good sir

**Reply by isRyven** [reply]
Seems working fine now, lgtm!

**Reply by rmarquis** [reply]
> Alright, I did the required change for HUD icons.
> 
> About the construction box on Braundorf, I can't reproduce the issue
> ![image](https://user-images.githubusercontent.com/5512525/121508375-d0a37880-c9e5-11eb-8fe5-22fe566f6a14.png)
> 
> Any tips ?

Unrelated, but you might want to fix that building and location labels overlap. It used to be fine afaik, but stuff might have been moved a bit.

**Reply by Aranud** [reply]
Oh, true. I will have a look.

**Reply by Aranud** [reply]
That overlap is normal and only appear when you start constructing. My screenshot is missleading because the location labels start to disappear when the building label appeared.

Just a bad timing screenshot :)

**Reply by rmarquis** [reply]
All right then!

**Comment by rafal1137** [comment]
I have reached a roadblock with Android part especially with `git_describe` command inside `ETLVersion.cmake`. Rest of the Android is fine except few issues with Android 10 Version.

**Comment by Aranud** [comment]
If everyone is ok, we should freeze to the current commit and start a 1-2 week test before releasing 2.78.

Anything to do before freeze ?

**Reply by rmarquis** [reply]
#1650 looks like an easy fix. Otherwise, current code can be frozen.

**Reply by Aranud** [reply]
As discusted on discord : 

Freeze from 23rd until the end of month and release on the 30th June

**Reply by rmarquis** [reply]
Hmm... maybe the new release is somehow late? To be fair, nobody mentioned the year :P

**Reply by Aciz** [reply]
Yeah it's late :D but to be fair, it would have been a bad idea to do a release at the end of June, there have been so many issues being resolved for the past month or two, that it would have been a bad release really.

**Reply by rmarquis** [reply]
True, but realistically the code will never be "good enough" for a release. It's the never ending balance between doing a stable enough "rolling release" approach and never releasing anything at all.

Let's say, what prevents the project to make a new release *now*? Any blocker or really huge bug?

**Reply by Aciz** [reply]
Personally I'd like to have the client detection of 2.60b servers for the patch prediction error done for 2.78, thats the only major issue really that I can think of, past week or so we have fixed the big game breaking bugs that have been there for a while though. But these recent changes should of course be tested properly first before a release (TM scrim servers are great for this, there's constant activity there and the environment is really controlled, unlike TM public), so rolling out the recent changes there for a while first before a release is a good idea.

Also there's this thing that I discovered literally 20 minutes ago, where self-killing above a non-axial ground plane while in air places your corpse underground :D not related to the recent changes to corpse handling, this is present all the way to 2.77.1 (didn't test earlier versions).

**Reply by ryzyk-krzysiek** [reply]
We've had a discussion like a week ago about what we should fix before release, most of it got fixed in that week. My list of things I wanted to see fixed for sure before release is now empty (maybe 1 more thing with server crashing), if everything works fine with recent changes which will be tested even more at scrim servers the release shouldn't be getting delayed by much.

**Reply by Aciz** [reply]
All in all there should definitely be a freeze time right now, only bug fixes to master.

**Reply by isRyven** [reply]
There is still a bug, related to body disappearing on gibbing (or tapout), to fix. And that dynamite chaining thingy that removes unarmed dynamites even with default cvar setting.

**Reply by rmarquis** [reply]
> And that dynamite chaining thingy that removes unarmed dynamites even with default cvar setting.

Do you mean the PR in #1716? Is there any other issue left before merging? I'd be in favour of having the feature enabled by default, but I might have missed some other side effects.

Freeze time starting now sounds good, for a release around 1st of October?

**Reply by Aciz** [reply]
Sounds good to me!

> There is still a bug, related to body disappearing on gibbing (or tapout), to fix. 

It's very likely this was fixed in #1717 though it's hard to verify since no one really knows a consistent method of reproducing this.

**Reply by rafal1137** [reply]
From my side I will add a merge of 2 repo's Android and main for better maintaining `things` that way.

**Reply by isRyven** [reply]
> Do you mean the PR in #1716? Is there any other issue left before merging? I'd be in favour of having the feature enabled by default, but I might have missed some other side effects.
> 

Yeah, I don't know if you intended it this way initially, but i think it's on by default. Not sure whether it's a bug or something else.

**Reply by rmarquis** [reply]
Seems unintended. Might have been introduced with the dyna chaining free commit (41d0b066ffbc15773160e48d1ea2c0fab7b352f6). Will have a closer look soon.

**Reply by ryzyk-krzysiek** [reply]
Ah so there is yet another check in `G_ChainFree`, god damn I'm blind..

**Reply by rmarquis** [reply]
FYI, I moved most ticket in a new 2.79 milestone. Only left are those that are afaik on-going for 2.78, and some that might be already fixed but require feedback. Feel free to move 2.79 tickets back to 2.78 milestone if misplaced.

**Comment by Aranud** [comment]
@rmarquis are we ready ? :)

**Reply by rmarquis** [reply]
I'll let you adjust the `VERSION.txt` file and tag the release.

Congratulation and thank you to all people involved!

**Reply by rafal1137** [reply]
Another Release Has Been Shiiped! 
Congratulations for everyone who did participated.
This Discussion Subject Can be Closed :)

**Reply by Aciz** [reply]
Thank you to everyone for your hard work, it's been a blast. Onwards we go!
